src/main/webapp/resources/js/pages/admin/components/statistics/AdvancedStatistics.jsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * This file renders the Advanced Statistics component
+ * which displays statistics in charts
+ */
+
+import React, { useContext, useEffect, useState } from "react";
+import { Bar, Column, Line, Pie } from "@ant-design/charts";
+import { Card, Form, PageHeader } from "antd";
+import { TimePeriodSelect } from "./TimePeriodSelect";
+import {
+  AdminStatisticsContext
+} from "../../../../contexts/statistics-context";
+
+import {
+  chartTypes,
+  defaultChartType,
+  defaultTimePeriod,
+  statisticTypes
+} from "../../statistics-constants";
+
+import { SPACE_LG, SPACE_MD } from "../../../../styles/spacing";
+import { ChartTypeButtons } from "./ChartTypeButtons";
+import { getChartConfiguration } from "../../chart-config";
+import { useNavigate } from "@reach/router";
+import { setBaseUrl } from "../../../../utilities/url-utilities";
+
+export default function AdvancedStatistics({ statType }) {
+  const {
+    adminStatisticsContext,
+    getUpdatedStatsForStatType
+  } = useContext(AdminStatisticsContext);
+
+  const [chartType, setChartType] = useState(defaultChartType);
+  const [form] = Form.useForm();
+  const navigate = useNavigate();
+
+  const components = {
+    [chartTypes.BAR]: Bar,
+    [chartTypes.COLUMN]: Column,
+    [chartTypes.LINE]: Line,
+    [chartTypes.PIE]: Pie,
+  };
+
+  useEffect(() => {
+    setChartType(defaultChartType);
+    form.setFieldsValue({
+      "time-period": defaultTimePeriod,
+    });
+  }, [statType]);
+
+  function updateTimePeriod(currTimePeriod) {
+    getUpdatedStatsForStatType(statType, currTimePeriod);
+  }
+
+  const chartTitle =
+    statType === statisticTypes.ANALYSES
+      ? "Number of Analyses Ran"
+      : statType === statisticTypes.PROJECTS
+      ? "Number of Projects Created"
+      : statType === statisticTypes.SAMPLES
+      ? "Number of Samples Created"
+      : statType === statisticTypes.USERS
+      ? "Number of Users Created"
+      : null;
```

Comment:
This complex ternary statements are overly complicated.  Why not just create an object:

```javascript
const TITLES = {
    [statisticTypes.ANALYSES]: "Number of Analyses Ran",
    [statisticTypes.PROJECTS]: "Number of Projects Created",
    [statisticTypes.SAMPLES]: "Number of Samples Created",
    [statisticTypes.USERS]: "Number of Users Created",
  };
```

Then in you JSX:

```javascript
title={TITLES[chartType]}
```

I would do the same thing with the large ternary below as well.

Reply:
Updated both blocks of ternary statements in 6c0aba73217bab99d94702474de6064bf84e81c1